File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field


class GenerateTaglinesRequest(BaseModel):
    product: str
    audience: str = ""
    tone: str = "professional"
    count: int = Field(default=5, ge=1, le=10)


class GenerateTaglinesResponse(BaseModel):
    taglines: list[str]


class GenerateImageRequest(BaseModel):
    prompt_context: str
    artifact_type: str = "poster"
    tone: str = "professional"
    style: str = "modern minimal"
    aspect_ratio: str = "1:1"


class GenerateImageResponse(BaseModel):
    image_url: str
    prompt_used: str


class StoryboardFrame(BaseModel):
    frame_number: int
    duration_seconds: int
    text_overlay: str
    visual_description: str
    transition: str


class GenerateStoryboardRequest(BaseModel):
    topic: str
    key_message: str
    product: str = ""
    tone: str = "professional"


class GenerateStoryboardResponse(BaseModel):
    frames: list[StoryboardFrame]

File: frontend/src/types/ai.ts
export interface GenerateTaglinesRequest {
  product: string;
  audience: string;
  tone: string;
  count?: number;
}

export interface GenerateTaglinesResponse {
  taglines: string[];
}

export interface GenerateImageRequest {
  prompt_context: string;
  artifact_type: string;
  tone: string;
  style?: string;
  aspect_ratio: string;
}

export interface GenerateImageResponse {
  image_url: string;
  prompt_used: string;
}

export interface StoryboardFrame {
  frame_number: number;
  duration_seconds: number;
  text_overlay: string;
  visual_description: string;
  transition: string;
}

export interface GenerateStoryboardResponse {
  frames: StoryboardFrame[];
}
